transformers/template_rgb_plot.md

```diff
@@ -0,0 +1,57 @@
+Also, information on the contents and intent of each of the executable and dependency files is provided (dependency files are files that upstream components are dependent upon).
+
+## Goals
+The fundamental goal of this repository is to make it as easy as possible for plant science researchers to implement their algorithms for incorporation into processing pipelines.
+To that end, we provide a standard interface, and request minimal configuration on the part of the researcher/developer.
+
+We also provide a testing mechanism that can be used to prove the algorithm implementation before incorporating it into a processing pipeline.
+This testing mechanism can also be used to track down issues arising in an active processing pipeline.
+
+## Files
+This section provides technical information on the contents of the file found in the [repository](https://github.com/AgPipeline/template-rgb-plot).
+
+This section is not intended to provide information on all the files in the repository, or to replace or document what is implemented in the code.
+There are files in the repository that conform to convention, such as '.gitignore' and are documented elsewhere.
+It is expected that the contents of the files are sufficiently documented so that external documentation isn't needed; this is not to say that there isn't external documentation generated from the source files, only that it's not to be found here.
+
+### algorithm_rgb.py <a name="algorithm_rgb" />
+This is the main file for the algorithm implementation.
+Its intent is to keep everything that is specific to the algorithm in one place; to serve as the main entry point.
+
+However, for more complicated algorithms, it may be necessary to have additional files that support the algorithm.
+This is anticipated and acceptable.
+
+Note that the default Dockerfile that is [generated](#generate) copies all `.py` files, but not any other file types.
+
+## generate.py <a name="generate" />
+This executable python script creates the Dockerfile based upon the contents of the [algorithm_rgb.py](#algorithm_rgb) file.
+Additionally empty `requirements.txt` and `packages.txt` files are generated.
+These last two can be used to install Python modules and system packages needed by the algorithm into a Docker image.
+
+The raw Dockerfile contents are stored as a list of strings in this file.
+These strings are modified as needed when writing Dockerfile.
+The decision to use this approach, instead of using template files, was made to reduce the footprint of the repository, and to keep the Docker related information in one spot.
+
+The script, when run, first checks that expected top-level variables are defined in the algorithm_rgb.py file and that some of these variables are not empty.
+Depending upon what the intent of a variable is, it's possible to get away with keeping them empty in the algorithm_rgb.py file.
```

Comment:
this isn't clear ... what are example intents? and would it change the meaning to remove the phrase "get away with"?